lib/MusicBrainz/Server/Controller/WS/2/Genre.pm

```diff
@@ -25,6 +32,27 @@ sub base : Chained('root') PathPart('genre') CaptureArgs(0) { }
 # Nothing extra to load yet, but this is required by Role::Lookup
 sub genre_toplevel {}
 
+sub genre_all : Chained('base') PathPart('all') {
+    my ($self, $c) = @_;
+
+    $c->detach('method_not_allowed')
+        unless $c->req->method eq 'GET';
+
+    $c->detach('genre_browse')
+        if $c->stash->{linked};
```

Comment:
I think browsing would be handled by `genre_search` already once implemented, not `/all`.